core/DataAccess/LogQueryBuilder/JoinTables.php

```diff
@@ -125,6 +159,85 @@ public function sort()
         $this->exchangeArray($sorted);
     }
 
+    public function isTableJoinableOnVisit($tableToCheck)
+    {
+        $table = $this->getLogTable($tableToCheck);
+
+        if (empty($table)) {
+            return false;
+        }
+
+        if ($table->getColumnToJoinOnIdVisit()) {
+            return true;
+        }
+
+        if ($table->getColumnToJoinOnIdAction()) {
```

Comment:
is that one actually needed? or is it some edge case?

Reply:
I'll remove it. Guess I assumed that no one would define multiple methods to allow a join. But actually one could define a way to join on action and on visit. Not sure if that would make sense, but at least it would be possible then.